### DISCUSSION TITLE
Generating switch configs for non-eos devices

### DISCUSSION THREAD

**Discussion by Trailingslashes** [aristanetworks/avd#4589] [Q&A] [answered]
I'm wondering if it's possible to generate device configurations that aren't a part of my AVD fabric. Perhaps it's possible with PyAVD alone, but I'm not sure. When I checked avd.sh, the site said "Generation of device configuration," so I thought it might be possible to generate configs and documentation for Aruba/Comware, etc.

**Comment by carlbuchmann** [comment, marked as answer]
@Trailingslashes Yes, it is possible to generate configurations for other vendors.
The `eos_designs` role outputs a "structured configuration", which you can use to generate a configuration for another vendor (e.i. replace `eos_cli_config_gen` with `aruba_config_gen`).

Note that when generating the configuration, we rely on pushing it using a "config replace," so we don't have to consider the device's current running configuration state.

**Reply by Trailingslashes** [reply]
Are there any documentation or examples I can take a look at?
